vue/sbc-common-components/src/components/BaseAddress.vue

```diff
@@ -161,6 +161,7 @@ import CountriesProvincesMixin from '../mixins/countries-provinces-mixin'
   mixins: [ValidationMixin, CountriesProvincesMixin]
 })
 export default class BaseAddress extends Mixins(ValidationMixin, CountriesProvincesMixin) {
+  $v: any
```

Comment:
This should already be defined on the Vue instance (which is inherited via the Mixins) by the `validation` plugin helper (imported on line 149, used on line 168).

Lewis, does this actually solve the problem for you?

Reply:
The test unit in this project doesn't have any problem, but testing this component in other projects, such as Completing Party in bcrs-shared will have the bug above. I cannot tell whether this will solve the problem, but after defining `$v` in the beginning, the error is gone in [line 359](https://github.com/bcgov/sbc-common-components/blob/4efa1e6664dff321030eca7448a7d5af91c2f355/vue/sbc-common-components/src/components/BaseAddress.vue#L359).

Reply:
While editing Edit UI and looking at node_modules/sbc-common-components/src/components/BaseAddress.vue, `this.$v` has a type already (and `$invalid` has type boolean).

![image](https://user-images.githubusercontent.com/10002889/182479138-aa32d826-82b4-401e-86e2-c1c8b194d17c.png)

I'm hesitant to change anything without more info on what you're seeing (and why casting to `any` works).

Reply:
If it's a unit-test only error, perhaps something could be worked around in the unit test itself?

Reply:
> If it's a unit-test only error, perhaps something could be worked around in the unit test itself?

To be clear, are you exercising BaseAddress in a higher-level component? You shouldn't need to. You should only unit test what the higher-level component is doing. You can stub out BaseAddress and simulate events if needed.

Reply:
Yes. I was exercising BaseAddress in a higher-level component, which is the CompletingParty component from bcrs-shared. This component uses BaseAddress directly. The unit test will fail when I tried to use "mount" to observe the behavior of inputs for first, middle and last names. I've tried many other ways, but this error `Cannot read properties of undefined (reading $invalid)` always shows up. That's why there is no unit test for Completing Party in the bcrs-shared project.

Reply:
Well, if `this.$v` is undefined, declaring it as type=any won't help at all.

I appreciate you trying to add missing unit tests.

Maybe the validator package is missing in the shared project/component to test this?

Reply:
Try adding [this package](https://github.com/bcgov/sbc-common-components/blob/4efa1e6664dff321030eca7448a7d5af91c2f355/vue/sbc-common-components/package.json#L68) to [this component](https://github.com/bcgov/bcrs-shared-components/blob/main/src/components/completing-party/package.json).

Reply:
I was just trying to make sure components whichever I was working on are working as expected with unit tests. Completing Party in the bcrs-shared project doesn't have a unit test, and the same for other projects that use this component. If we are fine with this component without unit test, then I will not include this unit test in Edit UI and we can close this PR 🙂

Reply:
If you can't figure it out, please create a Tech Debt ticket so we can keep track of this issue and maybe resolve in the future. Thx.

Reply:
Okay. Will do. Thanks for the help. I'll close this PR.